dspy/retrieve/azureaisearch_rm.py

```diff
@@ -89,83 +153,161 @@ def __init__(
+        self.is_vector_search = is_vector_search
+        self.is_hybride_search = is_hybride_search
+        self.is_fulltext_search = is_fulltext_search
+        self.vector_filter_mode = vector_filter_mode
 
         super().__init__(k=k)
 
-    def azure_search_request(self,key_content: str,  client: SearchClient, query: str, top: int, semantic_ranker: bool, filter: str, query_language: str, query_speller: str, use_semantic_captions: bool, query_type: QueryType, semantic_configuration_name: str):
+    def azure_search_request(
+        self,
+        key_content: str,
+        client: SearchClient,
+        query: str,
+        top: int,
+        semantic_ranker: bool,
+        filter: str,
+        query_language: str,
+        query_speller: str,
+        use_semantic_captions: bool,
+        query_type: QueryType,
+        semantic_configuration_name: str,
+        is_vector_search: bool,
+        is_hybrid_search: bool,
+        is_fulltext_search: bool,
+        field_vector: str,
+        vector_filter_mode: VectorFilterMode,
+    ):
         """
         Search in Azure AI Search Index
         """
 
-        # TODO: Add Support for Vector Search And Hybride Search
-        if semantic_ranker:
-            results = client.search(search_text=query,
-                                    filter=filter,
-                                    query_type=query_type,
-                                    query_language = query_language,
-                                    query_speller=query_speller,
-                                    semantic_configuration_name=semantic_configuration_name,
-                                    top=top,
-                                    query_caption = (
-                                        'extractive|highlight-false'
-                                        if use_semantic_captions
-                                        else None
-                                    ),
-                                )
-        else:
-            results = client.search(search_text=query,top=top,filter=filter)
+        if is_vector_search:
+            vector_query = self.get_embeddings(query, top, field_vector)
+            if semantic_ranker:
+                self.check_sementic_configuration(semantic_configuration_name, query_type)
+                results = client.search(
+                    search_text=None,
+                    filter=filter,
+                    query_type=query_type,
+                    vector_queries=[vector_query],
+                    vector_filter_mode=vector_filter_mode,
+                    semantic_configuration_name=semantic_configuration_name,
+                    top=top,
+                )
+            else:
+                results = client.search(
+                    search_text=None,
+                    filter=filter,
+                    vector_queries=[vector_query],
+                    vector_filter_mode=vector_filter_mode,
+                    top=top,
```

Comment:
not familiar with the semantics here myself but does this require a `query_caption` parameter like the other conditions?

Reply:
Yes, but that is optional parameter and I have provided the setting for the same if you are using semantic reranker. Just missed.